From b933a950ff8415217cbd29a4b0d89eace17c8cff Mon Sep 17 00:00:00 2001
From: Emre Ucan <eucan@de.adit-jv.com>
Date: Wed, 19 Apr 2017 10:23:16 +0200
Subject: [PATCH] ilmControl: fix memory corruption at input listener

Allocate memory in the size of accepted_seat struct instead of
size of a pointer of the struct

Signed-off-by: Emre Ucan <eucan@de.adit-jv.com>
---
 ivi-layermanagement-api/ilmControl/src/ilm_control_wayland_platform.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ivi-layermanagement-api/ilmControl/src/ilm_control_wayland_platform.c b/ivi-layermanagement-api/ilmControl/src/ilm_control_wayland_platform.c
index 0664016..5b4ad45 100644
--- a/ivi-layermanagement-api/ilmControl/src/ilm_control_wayland_platform.c
+++ b/ivi-layermanagement-api/ilmControl/src/ilm_control_wayland_platform.c
@@ -934,7 +934,7 @@ input_listener_input_acceptance(void *data,
         return;
     }
 
-    accepted_seat = calloc(1, sizeof(accepted_seat));
+    accepted_seat = calloc(1, sizeof(*accepted_seat));
     if (accepted_seat == NULL) {
         fprintf(stderr, "Failed to allocate memory for accepted seat\n");
         return;
-- 
2.1.4

